RULES.md

```diff
@@ -1,536 +1,1037 @@
-  TripUsageValidator(GtfsTripTableContainer tripTable, GtfsStopTimeTableContainer stopTimeTable) {
-    this.tripTable = tripTable;
-    this.stopTimeTable = stopTimeTable;
-  }
-
-  @Override
-  public void validate(NoticeContainer noticeContainer) {
-    // Do not report the same trip_id multiple times.
-    Set<String> reportedTrips = new HashSet<>();
-    for (GtfsTrip trip : tripTable.getEntities()) {
-      String tripId = trip.tripId();
-      if (reportedTrips.add(tripId) && stopTimeTable.byTripId(tripId).isEmpty()) {
-        noticeContainer.addValidationNotice(new UnusedTripNotice(tripId, trip.csvRowNumber()));
-      }
-    }
-  }
-  /**
-   * A {@code GtfsTrip} should be referred to at least once in {@code GtfsStopTimeTableContainer}
-   * station).
-   *
-   * <p>Severity: {@code SeverityLevel.WARNING}
-   */
-  static class UnusedTripNotice extends ValidationNotice {
-    UnusedTripNotice(String tripId, long csvRowNumber) {
-      super(
-          ImmutableMap.of(
-              "tripId", tripId,
-              "csvRowNumber", csvRowNumber),
-          SeverityLevel.WARNING);
-    }
-  }
-}
-```  
- 
-Note that the notice ID naming conventions changed in `v2` to make contributions of new rules easier by reducing the likelihood of conflicting IDs during parallel development. Please refer to [MIGRATION_V1_V2.md](/docs/MIGRATION_V1_V2.md) for a mapping between v1 and v2 rules.
-
+This document lists all the notices that are emitted by this validator.\
+Note that the notice naming convention changed in `v2` to make contributions of new rules easier by reducing the likelihood of conflicting IDs during parallel development. Please refer to [MIGRATION_V1_V2.md](/docs/MIGRATION_V1_V2.md) for a mapping between v1 and v2 notices.\
+Note that some severities were modified in `v3` to solve discrepancies with the specification. Please refer to [MIGRATION_V2_V3.md](/docs/MIGRATION_V2_V3.md) for a mapping between v2 and v3 notices.
 <a name="definitions"/>
 
 ## Definitions
-Notices are split into three categories: `INFO`, `WARNING`, `ERROR`.
+### A Rule
+A part of the specification that is translated into code in the validator. A Rule will describe if a set of conditions is met or not. For example:
+  - In the specification: in the `stops.txt` file, the field `zone_id` is required if providing fare information using `fare_rules.txt` ([source in the spec](https://gtfs.org/schedule/reference/#stopstxt)).
+  - In the validator: this is translated into code in the file `StopZoneIdValidator.java` ([link to the validator file](https://github.com/MobilityData/gtfs-validator/blob/master/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopZoneIdValidator.java)).
+
+### A Notice
+What the Rule outputs if the conditions aren’t met. It is what the user will see in the validation report.
+- For example, the output of `StopZoneIdValidator.java` Rule is the Notice `StopWithoutZoneIdNotice`, and it has the notice code `stop_without_zone_id`. 
```

Comment:
Related to my below comment on having notice class names in this document - I think including `StopWithoutZoneIdNotice` here is potentially confusing. Why is the name of the notice different from the notice code? If `stop_without_zone_id` is what I see in the validation output (JSON and HTML) why do I need to know the `StopWithoutZoneIdNotice` name? Do I need to reference that somewhere else?

IMHO this file shouldn't contain a bunch of implementation details but should be focused on explaining the business logic of the validation rules. I think the exception is including the collapsed notice field descriptions, which we're doing to try to keep all the docs in the same place.  The fact that we're hiding it by default probably means that most users won't see it (which is fine).

Reply:
> IMHO this file shouldn't contain a bunch of implementation details
Interesting POV. I don't see this as implementation details, but a key element to the comprehension of the validator as a casual user. I will re-work this part to make it less technical details looking!